src/WebJobs.Extensions.DurableTask/Correlation/TelemetryActivator.cs

```diff
@@ -39,17 +43,40 @@ public TelemetryActivator(IOptions<DurableTaskOptions> options, INameResolver na
         /// </summary>
         public Action<ITelemetry> OnSend { get; set; } = null;
 
+        /// <inheritdoc/>
+        public ValueTask DisposeAsync()
+        {
+            return this.disposable?.DisposeAsync() ?? default;
+        }
+
         /// <summary>
         /// Initialize is initialize the telemetry client.
         /// </summary>
         public void Initialize(ILogger logger)
         {
-            this.SetUpDistributedTracing();
-            if (CorrelationSettings.Current.EnableDistributedTracing)
+            if (this.options.Tracing.DistributedTracingEnabled || this.options.Tracing.NewDistributedTracingEnabled)
             {
                 this.endToEndTraceHelper = new EndToEndTraceHelper(logger, this.options.Tracing.TraceReplayEvents);
-                this.SetUpTelemetryClient();
-                this.SetUpTelemetryCallbacks();
+                this.SetupTelemetryConfiguration();
+                if (this.options.Tracing.NewDistributedTracingEnabled)
+                {
+                    DurableTelemetryModule module = new DurableTelemetryModule();
+                    module.Initialize(this.telemetryConfiguration);
+                    this.disposable = module;
+                }
+                else
+                {
+                    this.SetUpDistributedTracing();
```

Comment:
Should we rename this method to something like `SetUpDeprecatedDistributedTracing()` to make it clear that this applies only to the old implementation?